packages/commonwealth/server/routes/deleteProfile.ts

```diff
@@ -0,0 +1,71 @@
+import type { Request, Response, NextFunction } from 'express';
+import type { DB } from '../models';
+
+export const Errors = {
+  NotLoggedIn: 'Not logged in',
+  NoProfileProvided: 'No profile provided in query',
+  AddressesStillLinked: 'Cannot delete profile with addresses',
+};
+
+const deleteProfile = async (
+  models: DB,
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  if (!req.user) {
+    return next(new Error(Errors.NotLoggedIn));
+  }
+
+  if (!req.body.profileId) {
+    return next(new Error(Errors.NoProfileProvided));
+  }
+
+  const { profileId } = req.body;
+
+  const profileToDelete = await models.Profile.findOne({
+    where: {
+      id: profileId,
+    },
+  });
+
+  const existingAddresses = await profileToDelete.getAddresses();
+
+  if (existingAddresses.length > 0) {
+    return next(new Error(Errors.AddressesStillLinked));
+  }
+
+  const existingProfiles = await req.user.getProfiles();
+  const newProfiles = existingProfiles.filter(
+    (p) => p.id !== parseInt(profileId, 10)
+  );
+
+  const updateProfileStatus = await models.Profile.destroy({
+    where: {
+      id: profileId,
+    },
+  });
+
+  const updateUserStatus = await models.User.update(
+    {
+      Profiles: newProfiles,
+    },
+    {
+      where: {
+        id: req.user.id,
+      },
+    }
+  );
+
+  if (!updateProfileStatus && !updateUserStatus) {
```

Comment:
If I'm not mistaken, we should use a try-catch here (and in similar error-handling situations), as these calls will throw on error? + then we can halt control flow on the first failure. Not sure though, please confirm if possible.

Reply:
That's a good point, will add a try-catch to each of these calls.